Added Spring suport to application. BIG PROBLEM (look at commit message)

MykytaKostiuk opened the pull request:
There is a BIG problem.
@Autowire doesn't work, but I can get the bean from context. O_o

(I disabled security for now)
start app and try to open "http://localhost:8080/books" - I recive: "HTTP Status 500 java.lang.NullPointerException"
Because BookServiceImpl didn't inject to BookService bookService in com.hillel.servlets.BookServlet.

Please look at next files:
Here I add spring support to app.
1.WEB-INF/web.xml

Here I config spring application context.
2.WEB-INF/applicationContext.xml

Here I try to inject bean
3. com.hillel.servlets.BookServlet:
@Autowired
@Qualifier("bookService") //optional
private BookService bookService;  - doesn't inject the BookServiceImpl

But it is a @Service:
@Service("bookService")
public class BookServiceImpl

In the BookServlet file look at 61-62 lines.
I can get BookServiceImpl from context in 62 line.

MykytaKostiuk committed:
Added Spring suport to application.
There is a BIG problem.
@Autowire doesn't work, but I can get the bean from context. O_o

(I disabled security for now)
start app and try to open "http://localhost:8080/books" - I recive: "HTTP Status 500 java.lang.NullPointerException"
Because BookServiceImpl didn't inject to BookService bookService in com.hillel.servlets.BookServlet.

Please look at next files:
Here I add spring support to app.
1.WEB-INF/web.xml

Here I config spring application context.
2.WEB-INF/applicationContext.xml

Here I try to inject bean
3. com.hillel.servlets.BookServlet:
@Autowired
@Qualifier("bookService") //optional
private BookService bookService;  - doesn't inject the BookServiceImpl

But it is a @Service:
@Service("bookService")
public class BookServiceImpl

In the BookServlet file look at 61-62 lines.
I can get BookServiceImpl from context in 62 line.

dinamio reviewed (changes requested):
This one is working for me
https://stackoverflow.com/questions/11843690/autowiring-in-servlet
Please, check

Problem was about servlet vs spring DI

MykytaKostiuk committed:
Added Spring suport to application.

Thanks for your advice! It is really working.
News:
1. Added autocompletion to "author" and "country" inputs in add new book form.
2. Pressing on user in users page shows added books by him.
3. a few style changes

ajax connection property was moved to connection.js

dinamio commented on lab9/Kostiuk/bookstore/src/main/java/com/hillel/security/AuthHelper.java:
```diff
@@ -0,0 +1,51 @@
+    @Getter
+    @Setter
+    private static User currentUser;
+    private static byte[] secret = SecureRandom.getSeed(1024);
```
Does secret need to be the same from time to time?

dinamio commented on lab9/Kostiuk/bookstore/src/main/java/com/hillel/util/Utils.java:
```diff
@@ -0,0 +1,50 @@
+        return Optional.ofNullable(req.getParameter("id")).map(Long::valueOf);
+    }
+
+    public static boolean isNotEpmty(Collection collection) {
```
Looks like typo

dinamio reviewed (approved):
Looks good
